apps/pipelines/models.py

```diff
@@ -341,15 +340,20 @@ def invoke(
             return ChatMessage(content=output)
 
     def _save_message_to_history(
-        self, session: ExperimentSession, message: str, type_: ChatMessageType, metadata: dict, tags: list[str] = None
+        self,
+        session: ExperimentSession,
+        message: str,
+        type_: ChatMessageType,
+        metadata: dict,
+        tags: list[str] = None,
     ) -> ChatMessage:
         chat_message = ChatMessage.objects.create(
             chat=session.chat, message_type=type_.value, content=message, metadata=metadata
         )
 
         if tags:
-            for tag in tags:
-                chat_message.add_system_tag(tag, TagCategories.BOT_RESPONSE)
+            for tag_value, category in tags:
+                chat_message.add_system_tag(tag_value, category or "")
```

Comment:
I think we should rename this method to `create_and_add_tag` (`add_tag` is already taken) and only if the category is not empty we can mark it as a system tag.